src/transaction/transaction.cpp

```diff
@@ -65,10 +70,16 @@ void Transaction::pushCatalogEntry(CatalogSet& catalogSet, CatalogEntry& catalog
     case CatalogEntryType::TYPE_ENTRY: {
         KU_ASSERT(
             catalogEntry.getType() == CatalogEntryType::DUMMY_ENTRY && catalogEntry.isDeleted());
+        if (newCatalogEntry->hasParent()) {
```

Comment:
Is this for sequence? I'm not following sequence for now. Maybe you can just walk me through this line of change tomorrow.